### PR TITLE
Bokeh DataTable output of status.json http endpoint

### PR DESCRIPTION
**Pull request by canavandl**
potential UI element for distributed:

screen shot 2016-03-16 at 5 55 00 pm

Notebook code to run:
```python
from distributed import Executor, progress

e = Executor('10.20.20.182:8786')
e

from distributed.diagnostics.status_monitor import Status_Monitor
Status_Monitor(('10.20.20.182', 9786))

import dask.array as da
x = da.random.normal(10, 0.1, size=(1000000000,), chunks=(1000000,))
y = (x + 1).mean()
future = e.compute(y)
```

### CODE REVIEW

**Comment by canavandl** [comment]
ping @mrocklin 

I don't have much experience writing tornado code, so feel free to nitpick. If the OO structure is ok w/ you, i'll go ahead and try to add plots tonight/tomorrow.

File: distributed/diagnostics/status_monitor.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    addr: tuple, optional
+        (ip, port) of scheduler.  Defaults to scheduler of recent Executor
+    interval: Number, optional
+        Interval between updates.  Defaults to 1s
+    """
+    def __init__(self, addr=None, interval=1000.00):
+        if addr is None:
+            scheduler = default_executor().scheduler
+            if isinstance(scheduler, rpc):
+                addr = (scheduler.ip, scheduler.port) # doesn't work
```
Comment (mrocklin) [inline distributed/diagnostics/status_monitor.py]: Yeah, this port is for the TCP server, which we could also start using if performance becomes an issue.

File: distributed/diagnostics/status_monitor.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+        This opens a connection to the scheduler, sends it a function to run
+        periodically, streams the results back and uses those results to update
+        the bokeh figure
+        """
+
+        try:
+            response = yield self.client.fetch('http://%s:%d/status.json' % (ip, port))
+        except Exception as e:
+            print(e)
```
Comment (mrocklin) [inline distributed/diagnostics/status_monitor.py]: In this case we probably want to return or raise (or just let the exception percolate up)

**Comment by mrocklin** [comment]
This all looks fairly sane to me from a Tornado perspective.

File: distributed/diagnostics/status_monitor.py (hunk 1 above)
Comment (canavandl) [inline distributed/diagnostics/status_monitor.py]: I'll make the change. Re: performance issues, I'd assume we'd hit bokeh render cycle limits before http becomes slow enough to consider tcp.

File: distributed/diagnostics/status_monitor.py (hunk 2 above)
Comment (canavandl) [inline distributed/diagnostics/status_monitor.py]: I agreed. I'll remove

**Comment by canavandl** [comment]
In terms of building an application, it's be possible to add to the status_application.py module.

If it gets to the point of wanting a "real" web app, you can embed the ui components using this `autoload_server` [doc link](http://bokeh.pydata.org/en/latest/docs/user_guide/embed.html#server-data). Doing so would require a bokeh server (tornado server) and web server running.

**Comment by canavandl** [comment]
here is the data tables with a memory-per-worker plot (need too add labels)

screen shot 2016-03-21 at 10 49 29 am

**Comment by mrocklin** [comment]
Is it normal to see many client connections when I'm logged in on a single browser page?

```
INFO:bokeh.command.subcommands.serve:Starting Bokeh server on port 5006 with applications at paths ['/status_application']
ERROR:/home/mrocklin/Software/anaconda/lib/python3.4/site-packages/bokeh/core/validation/check.py:E-1000 (COLUMN_LENGTHS): ColumnDataSource column lengths are not all the same: ColumnDataSource, ViewModel:ColumnDataSource, ref _id: e3dc69bc-2126-4d1d-9c90-d5da258c5aa6
INFO:bokeh.server.views.ws:WebSocket connection opened
DEBUG:bokeh.server.views.ws:Receiver created for Protocol('1.0')
DEBUG:bokeh.server.views.ws:ServerHandler created for Protocol('1.0')
INFO:bokeh.server.views.ws:ServerConnection created
DEBUG:bokeh.server.session:Sending pull-doc-reply from session '7nHNKcpQed88HwQFWNDZHTr10myOt3Jc3jpY6vU6HVin'
DEBUG:bokeh.server.tornado:[pid 13650] 1 clients connected
DEBUG:bokeh.server.tornado:[pid 13650] 1 clients connected
DEBUG:bokeh.server.tornado:[pid 13650] 1 clients connected
DEBUG:bokeh.server.tornado:[pid 13650] 1 clients connected
DEBUG:bokeh.server.tornado:[pid 13650] 1 clients connected
DEBUG:bokeh.server.tornado:[pid 13650] 1 clients connected
DEBUG:bokeh.server.tornado:[pid 13650] 1 clients connected
DEBUG:bokeh.server.tornado:[pid 13650] 1 clients connected
DEBUG:bokeh.server.tornado:[pid 13650] 1 clients connected
DEBUG:bokeh.server.tornado:[pid 13650] 1 clients connected
DEBUG:bokeh.server.tornado:[pid 13650] 1 clients connected
DEBUG:bokeh.server.tornado:[pid 13650] 1 clients connected
DEBUG:bokeh.server.tornado:[pid 13650] 1 clients connected
DEBUG:bokeh.server.tornado:[pid 13650] 1 clients connected
DEBUG:bokeh.server.tornado:[pid 13650] 1 clients connected
DEBUG:bokeh.server.tornado:[pid 13650] 1 clients connected
DEBUG:bokeh.server.tornado:[pid 13650] 1 clients connected
DEBUG:bokeh.server.tornado:[pid 13650] 1 clients connected
DEBUG:bokeh.server.tornado:[pid 13650] 1 clients connected
DEBUG:bokeh.server.tornado:[pid 13650] 1 clients connected
DEBUG:bokeh.server.tornado:[pid 13650] 1 clients connected
DEBUG:bokeh.server.tornado:[pid 13650] 1 clients connected
DEBUG:bokeh.server.tornado:[pid 13650] 1 clients connected
DEBUG:bokeh.server.tornado:[pid 13650] 1 clients connected
DEBUG:bokeh.server.tornado:[pid 13650] 1 clients connected
DEBUG:bokeh.server.tornado:[pid 13650] 1 clients connected
DEBUG:bokeh.server.tornado:[pid 13650] 1 clients connected
DEBUG:bokeh.server.tornado:[pid 13650] 1 clients connected
DEBUG:bokeh.server.tornado:[pid 13650] 1 clients connected
DEBUG:bokeh.server.tornado:[pid 13650] 1 clients connected
DEBUG:bokeh.server.tornado:[pid 13650] 1 clients connected
DEBUG:bokeh.server.tornado:[pid 13650] 1 clients connected
DEBUG:bokeh.server.tornado:[pid 13650] 1 clients connected
DEBUG:bokeh.server.tornado:[pid 13650] 1 clients connected
DEBUG:bokeh.server.tornado:[pid 13650] 1 clients connected
DEBUG:bokeh.server.tornado:[pid 13650] 1 clients connected
DEBUG:bokeh.server.tornado:[pid 13650] 1 clients connected
DEBUG:bokeh.server.tornado:[pid 13650] 1 clients connected
DEBUG:bokeh.server.tornado:[pid 13650] 1 clients connected
DEBUG:bokeh.server.tornado:[pid 13650] 1 clients connected
DEBUG:bokeh.server.tornado:[pid 13650] 1 clients connected
DEBUG:bokeh.server.tornado:[pid 13650] 1 clients connected
DEBUG:bokeh.server.tornado:[pid 13650] 1 clients connected
DEBUG:bokeh.server.tornado:[pid 13650] 1 clients connected
DEBUG:bokeh.server.tornado:[pid 13650] 1 clients connected
DEBUG:bokeh.server.tornado:[pid 13650] 1 clients connected
DEBUG:bokeh.server.tornado:[pid 13650] 1 clients connected
DEBUG:bokeh.server.tornado:[pid 13650] 1 clients connected
DEBUG:bokeh.server.tornado:[pid 13650] 1 clients connected
DEBUG:bokeh.server.tornado:[pid 13650] 1 clients connected
```

**Comment by canavandl** [comment]
IIRC - I think that logging statement is printing the number of current sessions in a sleep/awake loop. The number of clients should increment with new sessions (maybe try another browser tab and see if it goes to 2?)

**Comment by mrocklin** [comment]
Ah, ok.  I thought it happened every time a new client connected

**Comment by mrocklin** [comment]
Two more questions:  

1.  What is the right way to have several active plots on a page?  Do we want to split up the `Status_Monitor` object to multiple things?
2.  What is the right way to host several pages/views?  Do we have a separate directory with many `foo_application.py` files and point bokeh server to that?

**Comment by mrocklin** [comment]
What is the best way to test this?

**Comment by mrocklin** [comment]
This was merged in a separate PR.  Closing.

**Pull request closed without merging**